### PR TITLE
Added ForkManager and tests.

### PR DESCRIPTION
**Pull request by VolodymyrBabaiev**
Introduced a ForkMaanger. The mechanism that allows getting Mainchain or Sidechain for specified MainchainHeight or SidechainConsensusEpochNumebr.

Description of this feature can be found on SDK Notion - https://www.notion.so/horizen-labs/Fork-Manager-c28433d6d0ec4ef8adb9dc14c94e99e8

### CODE REVIEW

File: sdk/src/main/scala/com/horizen/fork/ForkConfigurator.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.horizen.fork
+
+import scala.util.{Failure, Success, Try}
+
+abstract class ForkConfigurator {
+  def getBaseSidechainConsensusEpochNumbers():scConsensusEpochNumber
```
Comment (i-Alex) [inline sdk/src/main/scala/com/horizen/fork/ForkConfigurator.scala:6]: I think that the `BaseFork` must always refer to all `0` epoch numbers.
So we can set this method as final returning the `scConsensusEpochNumber(0,0,0)`.
For example, OpenStakeTx and BT limit are going to be a part of `SidechainFork1`

**Review by MarcoOl94 — APPROVED** [review]

File: sdk/src/main/scala/com/horizen/fork/ForkManager.scala [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        }
+
+        if (mainchainForks.isEmpty)
+            throw new RuntimeException("MainchainForks list is empty")
+
+        val mcForksIter = mainchainForks.iterator
+        var mcFork: BaseMainchainHeightFork = null
+
+        networkName match {
+            case "regtest" => mcForksIter.takeWhile(fork => fork.heights.regtestHeight <= mainchainHeight).foreach(fork => {mcFork = fork})
```
[the remark this answers is not in the document]
Comment (VolodymyrBabaiev) [reply sdk/src/main/scala/com/horizen/fork/ForkManager.scala:26]: takeWhile returns an Iterator. It doesn't have a method last.

[the remark this answers is not in the document]
Comment (i-Alex) [reply sdk/src/main/scala/com/horizen/fork/ForkManager.scala:26]: I got it. Them fine to me.

**Review by i-Alex — APPROVED** [review]
Good!

**Pull request merged**